compiler/backend/src/org/jetbrains/kotlin/codegen/ExpressionCodegen.java

```diff
@@ -1786,6 +1785,14 @@ private static boolean isLambdaVoidBody(@NotNull KtElement bodyExpression, @NotN
         return false;
     }
 
+    private static boolean isLambdaBody(@NotNull KtElement bodyExpression) {
```

Comment:
Maybe rewrite isLambdaVoidBody about to:

```
return isLambdaBody(bodyExpression) && Type.VOID_TYPE.equals(returnType);
```

Reply:
Good idea!